pallets/settlement/src/lib.rs

```diff
@@ -1348,14 +1354,24 @@ impl<T: Config> Module<T> {
         TransactionOutcome::Commit(Ok(Ok(())))
     }
 
-    fn prune_instruction(id: InstructionId) {
+    fn prune_instruction(id: InstructionId, executed: bool) {
         let legs = InstructionLegs::drain_prefix(id).collect::<Vec<_>>();
         let details = <InstructionDetails<T>>::take(id);
         VenueInstructions::remove(details.venue_id, id);
         <InstructionLegStatus<T>>::remove_prefix(id, None);
         InstructionAffirmsPending::remove(id);
         AffirmsReceived::remove_prefix(id, None);
 
+        if executed {
+            <InstructionStatuses<T>>::mutate(id, |d| {
+                *d = InstructionStatus::Success(System::<T>::block_number())
+            });
+        } else {
+            <InstructionStatuses<T>>::mutate(id, |d| {
+                *d = InstructionStatus::Rejected(System::<T>::block_number())
+            });
```

Comment:
Don't need to use `mutate` here, since the old value isn't used.  Can just use `insert` to change the instruction status.  Using `insert` will save a storage read.